pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/BaseTaskExecutor.java

```diff
@@ -56,4 +59,15 @@ protected Schema getSchema(String tableName) {
     Preconditions.checkState(schema != null, "Failed to find schema for table: %s", tableName);
     return schema;
   }
+
+  protected long getSegmentCrc(String tableNameWithType, String segmentName) {
+    TableType tableType = TableNameBuilder.getTableTypeFromTableName(tableNameWithType);
+    SegmentZKMetadata segmentZKMetadata = tableType == TableType.OFFLINE ? ZKMetadataProvider
+        .getOfflineSegmentZKMetadata(MINION_CONTEXT.getHelixPropertyStore(), tableNameWithType, segmentName)
+        : ZKMetadataProvider
+            .getRealtimeSegmentZKMetadata(MINION_CONTEXT.getHelixPropertyStore(), tableNameWithType, segmentName);
+    Preconditions.checkState(segmentZKMetadata != null, "Failed to find segment metadata for table: %s, segment: %s",
```

Comment:
Do not add this as pre-condition. Just log a warning and return success. If the metadata is not there, it is likely that the segment has been deleted. If you throw exception, helix will keep retrying this forever.

Reply:
+1 on that. Another drawback of just throwing an exception here is that the task status would be left unchanged without proper cleanup.

Reply:
done